src/lookup/lookup.ts

```diff
@@ -0,0 +1,52 @@
+import {
+  Catalog,
+  Dataset,
+  Distribution,
+  IRI,
+} from '@netwerk-digitaal-erfgoed/network-of-terms-catalog';
+import {QueryTermsService, Terms, TermsResult} from '../services/query';
+
+export class LookupService {
+  constructor(
+    private catalog: Catalog,
+    private queryService: QueryTermsService
+  ) {}
+
+  public async lookup(uris: IRI[], timeoutMs: number) {
+    const datasetToIris: Map<Dataset, IRI[]> = uris.reduce(
+      (datasetMap, iri) => {
+        const dataset = this.findDatasetForUriPrefix(iri);
+        if (dataset) {
+          const iris = [...(datasetMap.get(dataset) ?? []), iri];
+          datasetMap.set(dataset, iris);
+        }
+        return datasetMap;
+      },
+      new Map<Dataset, IRI[]>()
+    );
+
+    const lookups = <Promise<TermsResult>[]>[];
+    datasetToIris.forEach((iris, dataset) =>
+      lookups.push(this.query(dataset.distributions[0], iris, timeoutMs))
+    );
+
+    // TODO: decide whether we want to return results grouped by source or indexed by lookup URI.
```

Comment:
Good one. My guess is that it's not very useful to group results by source. The API user is probably unaware of the sources, and wants to get to the metadata of each term as quickly as possible (without having to iterate over the sources first). An index by URI makes sense. But the source information would then become a part of the term information?

Reply:
Exactly: that would mean that we cannot re-use `TermsQueryResult` as the output of both the search and the lookup query:

```graphql
  type TermsQueryResult {
    source: Source!
    terms: [Term]! @deprecated(reason: "Use 'result' instead")
    result: TermsResult!
  }
```

Instead, we need a new output type for the lookup query, something like `LookupResult` below.

Reply:
Yes, `[LookupResult]` looks good!

Reply:
Done!